src/wallet/spend.cpp

```diff
@@ -382,6 +382,15 @@ bool CWallet::AttemptSelection(const CAmount& nTargetValue, const CoinEligibilit
         results.emplace_back(std::make_tuple(waste, std::move(knapsack_coins), knapsack_value));
     }
 
+    // We include the minimum final change for SRD as we do want to avoid making really small change.
+    // KnapsackSolver does not need this because it includes MIN_CHANGE internally.
```

Comment:
I would prefer if all three selection methods would receive the actual `selection_target` and they would internally add `MIN_FINAL_CHANGE` where applicable. Whether or not an input parameter needs to be amended by a global constant makes more sense as an implementation detail of the called function than each callsite needing to be aware of it, and it would allow the function calls to the selection methods more alike.

I.e. I think adding the `MIN_FINAL_CHANGE` should happen inside of `SelectCoinsSRD(...)`

Reply:
Currently that is difficult to do with the way that we do the waste calculation. I think this can be done in the future, especially post #22019